### PR TITLE
Rename .to_json to .to_hash

### LINKED ISSUE #334: rename `to_json` instance method to `to_hash`
#### Issue Summary

All of the classes have a similar method called `to_json` that look like this:
```ruby
def to_json(*)
{
   'enable' => self.enable,
    'enable_text' => self.enable_text
}.delete_if { |_, value| value.to_s.strip == '' }
end
```
But if you look at this, its does not return _JSON_, its returning a compacted hash with each of the class's instance variables enumerated by key/val. 

```ruby
    37: def to_json(*)
    38:   ret = {
    39:     'email' => self.email,
    40:    'name' => self.name
    41:   }.delete_if { |_, value| value.to_s.strip == '' }
 => 43:   binding.pry
    44:   ret
    45: end

[1] pry(#<SendGrid::Email>)> ret
=> {"email"=>"test1@example.com", "name"=>"Example User"}
```

This makes the code somewhat unclear and difficult to understand.
This method and all places that call it should be replaced with the `to_hash` method.

### PR DESCRIPTION
**Pull request by anthonyfalzetti**
#Fixes #334 
### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation about the functionality in the appropriate .md file
- [ ] I have added in line documentation to the code I modified

### Short description of what this PR does:
- JSON.parse returns a Hash and to_json also returns a hash so for clarity I renamed .to_json to .to_hash.

If you have questions, please send an email to [Sendgrid](mailto:dx@sendgrid.com), or file a Github Issue in this repository.

### CODE REVIEW

**Comment by anthonyfalzetti** [comment]
I am working on fixing the code to make the tests pass. I will try to get this completed afterwork tonight

**Comment by anthonyfalzetti** [comment]
Ok fixed it. Please review and let me know if any changes need to be made

File: lib/sendgrid/helpers/mail/category.rb [hunk 1]
```diff
@@ -8,7 +8,15 @@ def initialize(name: nil)
...
     end
 
-    def to_json(*)
+    def name=(name)
+      @name = name
+    end
+
+    def name
+      @name
+    end
```
Comment (tachyons) [inline lib/sendgrid/helpers/mail/category.rb]: we already have `attr_accessor :name`. Why do we need this to write explicitly ?

Comment (anthonyfalzetti) [reply lib/sendgrid/helpers/mail/category.rb]: Yeah my bad. There were some merge conflicts and I think this was a left over.

**Pull request closed without merging**